ra/ra.go

```diff
@@ -2363,9 +2363,14 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 	// `sa.GetAuthorizations` returned an authorization that was very close to
 	// expiry. The resulting pending order that references it would itself end up
 	// expiring very soon.
-	// To prevent this we only return authorizations that are at least 1 day away
-	// from expiring.
-	authzExpiryCutoff := ra.clk.Now().AddDate(0, 0, 1)
+	// What is considered "very soon" changes depending on the profile's authz
+	// lifetime: if this profile's orders and authzs are usually valid for a long
+	// time, we say they should have at least one day still on the clock; if
+	// they're usually short lived, we say they should still have at least 1 hour.
+	authzExpiryCutoff := ra.clk.Now().Add(24 * time.Hour)
+	if profile.validAuthzLifetime < 24*time.Hour {
+		authzExpiryCutoff = ra.clk.Now().Add(time.Hour)
+	}
```

Comment:
Another possibility here would be something like
```suggestion
	authzExpiryCutoff := profile.orderLifetime / 8
```

This would reduce our current cutoff from 24hr to 21hr, and put shortlived authz reuse at 1hr. It would be more flexible in supporting other potential order and authz lifetimes introduced in the future, but might also seem a bit magical.

Reply:
Dividing by 8 is a super interesting way to scale the cutoff dynamically — I generally prefer it over hard-coding. One way to refine this idea further is to add a floor and a ceiling, so the cutoff remains reasonable across different authz lifetimes.

```go
// Set an authz cutoff of 12.5% of authz lifetime.
cutoffRatio := 0.125 
computedCutoff := profile.validAuthzLifetime * cutoffRatio
if computedCutoff < time.Hour {
    computedCutoff = time.Hour

} else if computedCutoff > 24*time.Hour {
    computedCutoff = 24 * time.Hour
}
authzExpiryCutoff := ra.clk.Now().Add(computedCutoff)
```

Thoughts?